File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double x=0;
        int i=0,j=0;
        double ans=0;
        while(j<k){
            ans+=nums[j];
            j++;
        }
        // cout<<ans<<endl;
        x=(double)(ans*1.0/k);
        // cout<<x<<endl;
        while(j<nums.size()){
            ans-=nums[i];
            i++;
            ans+=nums[j];
            x=max(x,(double)(ans*1.0/k));
            // cout<<(double)(ans*1.0/k)<<endl;
            j++;
        }
        return x;


    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        x=0
        i=0
        j=0
        ans=0
        while(j<k):
            ans+=nums[j]
            j+=1
        x=(ans*1.0)/k
        
        while(j<len(nums)):
            ans-=nums[i]
            i+=1
            ans+=nums[j]
            x=max(x,(ans*1.0/k))
            j+=1

        return x